web-app/src/utilities/dashboardEvents.js

```diff
@@ -49,3 +50,79 @@ export const getEventDuration = event => {
     return duration;
   }
 };
+
+/*
+  Booking Form Validation
+*/
+
+export const startDateValidation = formData => {
+  if (formData.isHalfday) {
+    formData.end = formData.start;
+  } else {
+    if (formData.start.isAfter(formData.end)) {
+      formData.end = formData.start;
+    }
+  }
+  return formData;
+};
+
+export const endDateValidation = formData => {
+  if (formData.isHalfday) {
+    formData.start = formData.end;
+  } else {
+    if (formData.end.isBefore(formData.start)) {
+      formData.start = formData.end;
+    }
+  }
+  return formData;
+};
+
+export const halfDayValidation = formData => {
+  formData.end = formData.start;
+  formData.eventTypeId = eventTypes.ANNUAL_LEAVE;
+  return formData;
+};
+
+/* 
+  Events Validation
+*/
+
+export const checkIfPastDatesSelected = start => {
+  const today = new moment();
+  return moment(start).isBefore(today);
+};
+
+export const checkIfDatesFallOnWeekend = (start, end) => {
+  if (moment(start).isoWeekday() > 5 && moment(end).isoWeekday() > 5) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+export const checkIfSelectedDatesOverlapExisting = (
+  events,
+  employeeId,
+  start,
+  end,
+  selectedEventId = null,
+) => {
+  const overlappingEvents = events.filter(event => {
+    const { employee, eventId } = event;
+    if (
+      employee &&
+      employee.employeeId === employeeId &&
+      selectedEventId !== eventId
+    ) {
+      var selectedDateRange = moment.range(
+        moment(start),
+        moment(end).endOf('day'),
+      );
+      var existingEvent = moment.range(moment(event.start), moment(event.end));
```

Comment:
I'd try to use let and const for variables.

Reply:
Of course. not sure how that slipped in there.